Feature: Add SANDBOX

tfredian opened the pull request:
This feature provides a SANDBOX setting which will disable the use
of the spawn function or the py() tdi function for security reasons.
The SANDBOX setting can be turned on using the MdsShr->MdsEnableSandbox()
function. Once enabled for the process it cannot be disabled. When
enabled the LibSpawn function will return and error and execution of the py()
function will also be disabled. You can now also add mdsip.hosts mapping rules
such as:

<match-string> | SANDBOX

and if the connection matches the string it will enable SANDBOX in the mdsip
process and use the nobody uid.

tfredian committed:
Feature: Add SANDBOX

This feature provides a SANDBOX setting which will disable the use
of the spawn function or the py() tdi function for security reasons.
The SANDBOX setting can be turned on using the MdsShr->MdsEnableSandbox()
function. Once enabled for the process it cannot be disabled. When
enabled the LibSpawn function will return and error and execution of the py()
function will also be disabled. You can now also add mdsip.hosts mapping rules
such as:

<match-string> | SANDBOX

and if the connection matches the string it will enable SANDBOX in the mdsip
process and use the nobody uid.

tfredian committed:
Fix: Repair bug in mdsshr/gen_messages.py

WhoBrokeTheBuild commented on mdsshr/Makefile.in:
```diff
@@ -34,6 +34,7 @@ SOURCES = \
         MdsGetSetShotId.c \
         MdsGetStdMsg.c \
         MdsPk.c \
+	MdsSandbox.c \
```
nit: Tab not spaces

WhoBrokeTheBuild commented on include/mdsshr.h:
```diff
@@ -90,6 +91,7 @@ extern int MdsFree1Dx(mdsdsc_xd_t *const dsc, void **const zone);
 extern int MdsMsg(const int status, const char *const fmt, ...);
 extern int MDSQueueEvent(const char *const event, int *const eventid);
 extern int MDSGetEventQueue(const int eventid, const int timeout, int *const data_len, char **const data);
+extern int MdsSandbox();
```
Maybe something like `MdsIsSandbox()`? or `MdsSandboxEnabled()`?

WhoBrokeTheBuild commented on python/MDSplus/mdsExceptions.py:
```diff
@@ -2211,7 +2211,7 @@ class TreeINVTAG(TreeException):
 class TreeNOPATH(TreeException):
   status=265392114
-  message="No 'treename'_path environment variable defined. Cannot locate tree files."
+  message="No 'treename'_path or default_tree_path environment variables defined. Cannot locate tree files."
```
Nice catch

WhoBrokeTheBuild reviewed (dismissed):
Looks great! Just the few nitpicks

joshStillerman commented:
Tab not spaces ?  I would have Said: space not tab.  We should decide on 
a coding standard.  And run lint


On 8/12/19 12:41 PM, Stephen Lane-Walsh wrote:
>
> *@WhoBrokeTheBuild* approved this pull request.
>
> Looks great! Just the few nitpicks
>
> ------------------------------------------------------------------------
>
> In mdsshr/Makefile.in 
> <https://github.com/MDSplus/mdsplus/pull/1759#discussion_r313017484>:
>
> > @@ -34,6 +34,7 @@ SOURCES = \
>           MdsGetSetShotId.c \
>           MdsGetStdMsg.c \
>           MdsPk.c \
> +	MdsSandbox.c \
>
> nit: Tab not spaces
>
> ------------------------------------------------------------------------
>
> In include/mdsshr.h 
> <https://github.com/MDSplus/mdsplus/pull/1759#discussion_r313017813>:
>
> > @@ -90,6 +91,7 @@ extern int MdsFree1Dx(mdsdsc_xd_t *const dsc, void **const zone);
>   extern int MdsMsg(const int status, const char *const fmt, ...);
>   extern int MDSQueueEvent(const char *const event, int *const eventid);
>   extern int MDSGetEventQueue(const int eventid, const int timeout, int *const data_len, char **const data);
> +extern int MdsSandbox();
>
> Maybe something like |MdsIsSandbox()|? or |MdsSandboxEnabled()|?
>
> ------------------------------------------------------------------------
>
> In python/MDSplus/mdsExceptions.py 
> <https://github.com/MDSplus/mdsplus/pull/1759#discussion_r313018199>:
>
> > @@ -2211,7 +2211,7 @@ class TreeINVTAG(TreeException):
>   
>   class TreeNOPATH(TreeException):
>     status=265392114
> -  message="No 'treename'_path environment variable defined. Cannot locate tree files."
> +  message="No 'treename'_path or default_tree_path environment variables defined. Cannot locate tree files."
>
> Nice catch
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub 
> <https://github.com/MDSplus/mdsplus/pull/1759?email_source=notifications&email_token=ABY5AZM5KNXMXUA7DDTNOPLQEGHCRA5CNFSM4IKOMBK2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCBJAV4A#pullrequestreview-273812208>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABY5AZOJ3A2HNJXJ25VTEKTQEGHCRANCNFSM4IKOMBKQ>.
>

WhoBrokeTheBuild commented:
Yeah good call Josh, this is what I meant:
"Use spaces like the lines around it, instead of a tab" 
But +1 to coding standard and linting

tfredian committed:
Fix: Change MdsSandbox to MdsSandboxEnabled

For clarity change MdsSandbox function to MdsSandboxEnabled and
use consistent indentation in Makefile
